python/ray/_raylet.pyx

```diff
@@ -1333,6 +1333,8 @@ cdef class CoreWorker:
                         CRayObject(data, metadata, contained_object_refs),
                         contained_object_ids, c_object_id))
             else:
+                if os.name == "nt":
```

Comment:
I think this check should be removed, as far as I know the code can go through this even for single node Ray (Ray on a single node still uses multiple processes to do the work, and it also uses TCP to communicate between these processes).